var/spack/repos/builtin/packages/axl/package.py

```diff
@@ -64,7 +65,7 @@ def cmake_args(self):
             args.append("-DAXL_LINK_STATIC=ON")
         args.append("-DWITH_KVTREE_PREFIX=%s" % self.spec['kvtree'].prefix)
 
-        apis = self.spec.variants['async_api'].value.split(',')
+        apis = list(self.spec.variants['async_api'].value)
```

Comment:
What data type is it previously?

Reply:
a tuple. we end up calling "remove" then doing something with everything that remains in the list... so a tuple didn't work. I'm not sure why this worked previously, but now is failing.

Reply:
@alalazo I've seen multiple PRs lately that required changes to `spec.variants['...'].value` (also see #8872). Did something change recently? How many other packages are broken??

Reply:
@gonsie Do you mean `type(apis) == tuple`?

Reply:
@alalazo That's what it sounds like based on https://github.com/spack/spack/pull/8872#discussion_r208641673.

Reply:
It looks like the setter for variants has enforced that it be a tuple since https://github.com/spack/spack/commit/4f8c7d57eb3839ea866a9fbfb55f9a44af99d6c0 (11 months ago).

If I check out https://github.com/spack/spack/blame/develop/var/spack/repos/builtin/packages/axl/package.py and try:

```
$ spack-python
> x = Spec('axl')
> x.concretize()
> x.variants['async_api'].value
('daemon', ) 
```

The result is a tuple (likewise if I explicitly set the variant value, either with 1 value or multiple). So I don't think Spack has changed unexpectedly.